internal/pkg/deploy/cloudformation/stack/workload.go

```diff
@@ -101,7 +101,8 @@ func (i ECRImage) GetLocation() string {
 	return fmt.Sprintf("%s:%s", i.RepoURL, "latest")
 }
 
-type addons interface {
+// NestedStackConfigurer configures a nested stack that deploys addons.
+type NestedStackConfigurer interface {
```

Comment:
The reason why I exported this interface is because it is a part of the input for the constructors of `Env`, `LBWS`, `RDWS` etc.

Reply:
nit: What do you think of renaming the the interface to
```suggestion
type StringSerializer interface {
```
Since it seem to be more flexible than just for nested stacks

Reply:
I think the intended usage of `Parameters() (string, error)` is pretty specific to nested stacks 🤔  Only nested stacks would need to render for example, the below snippet
```yaml
Extra1: !Ref Somthing
Extra2: !Ref SomethingElse
```
goes [here](https://github.com/aws/copilot-cli/blob/534e4296060a40dd746420154fc891062e6634fa/internal/pkg/template/templates/environment/cf.yml#L659).

For a regular stack, the values of parameters are passed in API calls, not in the plain string template.

---

Instead of `Parameters() (string, error)`, maybe a better name for the function would be `ExtraParameters() (string, error)` - because it returns only the extra parameters:
```yaml
Extra1: !Ref Somthing
Extra2: !Ref SomethingElse
```
Whereas, a function named `Parameters() (string, error)` (or better, [`SerializedParameters() (string, error)`](https://github.com/aws/copilot-cli/blob/821c7086ff66c7c1760c0f5027b4a1a6fe050746/internal/pkg/deploy/cloudformation/stack/worker_svc.go#L180)) should return both the extra parameters and the Copilot-required ones:
```yaml
{
  "Parameters": {
    "App": "!Ref AppName",
    "Env": "!Ref Name",
    "Extra1": "!Ref Something",
    "Extra2": "!Ref SomethingElse",
  },
}
```

The name `StringSerializer` sounds like it should have `Template()(string, error)` and `SerializedParameters() (string, error)`; an interface that contains `Template()(string, error)` and `ExtraParameters() (string, error)` should probably be named specifically after "nested stacks"

Reply:
Gotcha yea, then `ExtraParameters` would have been perfect 😭 If it's not too much work afterwards we can refactor it to that